dynatrace_extension/cli/create/extension_template/setup.py.template

```diff
@@ -1,27 +1,196 @@
+import json
 from pathlib import Path
+import re
 from setuptools import setup, find_packages
 
 
-def find_version() -> str:
-    version = "0.0.1"
-    extension_yaml_path = Path(__file__).parent / "extension" / "extension.yaml"
-    try:
-        with open(extension_yaml_path, encoding="utf-8") as f:
-            for line in f:
-                if line.startswith("version"):
-                    version = line.split(" ")[-1].strip("\"")
-                    break
-    except Exception:
-        pass
-    return version
-
-
-setup(name="%extension_name%",
-      version=find_version(),
-      description="%Extension_Name% python EF2 extension",
-      author="Dynatrace",
+class ExtensionMetadata():
+    '''
+    Collects and makes available all extension metadata available to populate
+    the setup() command when building the extension.
+    '''
+
+    def __init__(self):
+        '''
+        Initialize our internal data structures accordingly.
+        '''
+
+        self._activation = self.__read_activation()
+        self._extension = self.__read_extension()
+
+    def __read_activation(self):
```

Comment:
No need for double underscore here, this is an old python practice, a single underscore is enough to indicate a private method